tests/ignite/metrics/test_accuracy.py

```diff
@@ -0,0 +1,161 @@
+    y = torch.ones(4, 1).type(torch.LongTensor)
+    acc.update((y_pred, y))
+    assert isinstance(acc.compute(), float)
+    assert acc.compute() == 0.75
+
+
+def test_binary_compute_batch_images():
+    acc = Accuracy()
+
+    y_pred = torch.FloatTensor([[[0.3, 0.7],
+                                 [0.2, 0.8]]])
+    y = torch.ones(1, 2, 2).type(torch.LongTensor)
+    acc.update((y_pred, y))
+    assert isinstance(acc.compute(), float)
+    assert acc.compute() == 0.5
+
+    acc.reset()
+    y_pred = torch.FloatTensor([[[0.3, 0.7],
+                                 [0.9, 0.8]]])
+    y = torch.ones(1, 2, 2).type(torch.LongTensor)
+    acc.update((y_pred, y))
+    assert isinstance(acc.compute(), float)
+    assert acc.compute() == 0.75
+
+    acc.reset()
+    y_pred = torch.FloatTensor([[[0.3, 0.7],
+                                 [0.9, 0.8]],
+                                [[0.8, 0.3],
+                                 [0.9, 0.4]]])
+    y = torch.ones(2, 2, 2).type(torch.LongTensor)
+    acc.update((y_pred, y))
+    assert isinstance(acc.compute(), float)
+    assert acc.compute() == 0.625
+
+    acc.reset()
+    y_pred = torch.zeros(2, 1, 2, 2).type(torch.FloatTensor)
+    y = torch.ones(2, 2, 2).type(torch.LongTensor)
+    acc.update((y_pred, y))
+    assert isinstance(acc.compute(), float)
+    assert acc.compute() == 0.
+
+
+def test_categorical_compute():
+    acc = Accuracy()
+
+    y_pred = torch.eye(4)
+    y = torch.ones(4).type(torch.LongTensor)
+    acc.update((y_pred, y))
+    assert isinstance(acc.compute(), float)
+    assert acc.compute() == 0.25
+
+    acc.reset()
+    y_pred = torch.eye(2)
+    y = torch.ones(2).type(torch.LongTensor)
+    acc.update((y_pred, y))
+    assert isinstance(acc.compute(), float)
+    assert acc.compute() == 0.5
+
+
+def test_categorical_compute_batch_images():
+    acc = Accuracy()
+    y_pred = torch.zeros(2, 3, 2, 2)
+    y_pred[0, 1, :] = 1
+    y_pred[1, 2, :] = 1
+
+    y = torch.LongTensor([[[0, 1],
+                           [0, 1]],
+                          [[0, 2],
+                           [0, 2]]])
+
+    acc.update((y_pred, y))
+
+    assert isinstance(acc.compute(), float)
+    assert acc.compute() == 0.5
+
+    acc.reset()
+    y_pred = torch.zeros(2, 3, 2, 2)
+    y_pred[0, 1, :] = 1
+    y_pred[1, 2, :] = 1
+
+    y = torch.LongTensor([[[2, 1],
+                           [1, 1]],
+                          [[2, 2],
+                           [0, 2]]])
+
+    acc.update((y_pred, y))
+    assert isinstance(acc.compute(), float)
+    assert acc.compute() == 0.75
+
+
+def test_multi_label():
+    acc = Accuracy()
+
+    y_pred = torch.eye(4)
+    y = torch.Tensor([[1., 0., 1., 0.],
+                      [0., 1., 0., 0.],
+                      [0., 0., 1., 0.],
+                      [1., 0., 0., 1.]]).type(torch.LongTensor)
+
+    acc.update((y_pred, y))
+    assert acc.compute() == 0.875
+
+
+def test_ner_example():
```

Comment:
Could you please explain what this test is testing ?

Reply:
@vfdev-5 discussed below! this test is wrong.

Reply:
Finally, do we keep this test ?

Reply:
NER example we should keep because it is another type of problem users might encounter. I have removed any multi label tests.

Reply:
Actually what means NER ?

Reply:
Named Entity Recognition.